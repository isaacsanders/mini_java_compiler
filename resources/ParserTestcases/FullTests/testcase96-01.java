class Good1{
    public static void main (String[] ID){
	int i = ------------------------------------------------1 + 2 + 3 + -4 
	    + 5 / 6 * 7 - 8 * (2 * 3 + 4 * 5);
	boolean j = !!!!!!!!!!!!!!!!!!!!!!!!!!!!!(i <= 3 || 3 < 7 && i >= k);
    }
}