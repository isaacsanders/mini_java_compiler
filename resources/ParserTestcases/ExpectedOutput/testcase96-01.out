StmtList ::= 
Type ::= INT
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
UnaryExpr ::= MINUS UnaryExpr
Expr ::= UnaryExpr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr PLUS Expr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr PLUS Expr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
UnaryExpr ::= MINUS UnaryExpr
Expr ::= UnaryExpr
Expr ::= Expr PLUS Expr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr DIVIDE Expr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr TIMES Expr
Expr ::= Expr PLUS Expr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr TIMES Expr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr TIMES Expr
Expr ::= Expr PLUS Expr
ParenExpr	::=	LPAREN Expr RPAREN
LiteralExpr ::= ParenExpr
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr TIMES Expr
Expr ::= Expr MINUS Expr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
Type ::= BOOLEAN
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr LTE Expr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= INTEGER
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr LT Expr
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
LiteralExpr ::= ID
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
Expr ::= UnaryExpr
Expr ::= Expr GTE Expr
Expr ::= Expr AND Expr
Expr ::= Expr OR Expr
ParenExpr	::=	LPAREN Expr RPAREN
LiteralExpr ::= ParenExpr
CallExpr ::= LiteralExpr
UnaryExpr ::= CallExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
UnaryExpr ::= NOT UnaryExpr
Expr ::= UnaryExpr
Stmt ::= Type ID GETS Expr SEMI
StmtList ::= StmtList Stmt
MainClassDecl ::= ...
ClassDeclList ::= 
Program ::= MainClassDecl ClassDeclList
